src/FilesystemUtils/FilesystemUtils.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types = 1);
+
+namespace Keboola\DockerApplication\FilesystemUtils;
+
+use const DIRECTORY_SEPARATOR;
+use function array_shift;
+use function array_unshift;
+use function implode;
+use function rtrim;
+
+class FilesystemUtils
+{
+    /**
+     * Checks if a given path is inside a given directory
+     *
+     * @param string $path
+     * @param string $directory
+     * @return bool
+     */
+    public static function isPathInDirectory(string $path, string $directory): bool
+    {
+        $directory = self::addTrailingSlash(self::platformIndependentPath($directory));
+        $path = self::addTrailingSlash(self::platformIndependentPath($path));
+
+        return strpos($path, $directory) === 0;
+    }
+
+    /**
+     * Works like `os.path.join` in python. Joins multiple path segments,
+     * ensuring that there are no double slashes. Returns platform independent
+     * path.
+     *
+     * @param string ...$segments
+     * @return string
+     */
+    public static function pathFromSegments(string ...$segments): string
+    {
```

Comment:
je to na neco potreba? 1) zdvojeni lomitek afaik nevadi 2) pri tom jak se to vola staci akorat vedet jestli basedir konci lomitkem nebo ne, nikde jinde se tam stejne dvojite lomitko nemuze objevit

Reply:
Jak kdy. Třeba když si uděláš `'/data/' . '/tables/in'` a pak kontroluješ, jestli jsi ve správném podadresáři `substr($dir, '/data/tables/in')` tak to vadí.

Reply:
Více méně je to alternativa k `os.path.join` https://github.com/keboola/python-docker-application/blob/master/keboola/docker.py#L259

Reply:
no ten os.path.join je tam prave celkem naprd afaik a vsechno ostatni imho resi https://api.symfony.com/2.7/Symfony/Component/Finder/SplFileInfo.html#method_getRelativePath

Reply:
celkove me proste prijde, ze FileSystemUtils je NIH

Reply:
GetRelativePath je právě [jen getter toho, co mu dáš v konstruktoru](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Finder/SplFileInfo.php#L45). Takže se to nehodí na abstraktní práci s cestami, protože ta cesta musí existovat a pak to relative path vyplní `Finder`.

Reply:
Já jsem zkoušel jestli to nezvládnu nějak přes Symfony Filesystem + Finder nejdřív.

Reply:
Odebráno v https://github.com/keboola/php-docker-application/commit/797de5a21196c065e3558ed07815cce2f0ebbbce